src/Controllers/FrontEnd/StudentProfilePageController.php

```diff
@@ -32,7 +32,7 @@ public function __invoke(Request $request, Response $response, array $args): Res
         $params['applicant'] = $this->applicantModel->getApplicantById($params['id']);
 
         if ((!$params['applicant']) || !$params['applicant']->isStudentStage()) {
-             return $this->renderer->render($response, '404.phtml');
+             return $this->renderer->render($response->withStatus(404), '404.phtml');
```

Comment:
we should apply the status to the response before passing the response into render, just in case